contracts/governance/staking/StakedToken.sol

```diff
@@ -231,19 +228,22 @@ contract StakedToken is IStakedToken, GamifiedVotingToken {
 
         // 2. Deal with cooldown
         //      If a user is currently in a cooldown period, re-calculate their cooldown timestamp
-        uint256 nextCooldown = getNextCooldownTimestamp(
-            _amount,
-            _msgSender(),
-            balanceOf(_msgSender())
-        );
+        CooldownData memory oldCooldown = stakersCooldowns[_msgSender()];
         //      If we have missed the unstake window, or the user has chosen to exit the cooldown,
         //      then reset the timestamp to 0
         bool exitCooldown = _exitCooldown ||
-            block.timestamp > (nextCooldown + COOLDOWN_SECONDS + UNSTAKE_WINDOW);
+            block.timestamp > (oldCooldown.timestamp + COOLDOWN_SECONDS + UNSTAKE_WINDOW);
         if (exitCooldown) {
             stakersCooldowns[_msgSender()] = CooldownData(0, 0);
+            _exitCooldownPeriod(_msgSender());
```

Comment:
This line is not actually necessary, this will be picked up in the `_mintRaw` fn (see point 2. in mintRaw in GamifiedToken)

Reply:
If I remove this the `should end partial cooldown via staking` test fails. The `previousBalance` in the `DelegateVotesChanged` event goes from the correct 7,000 to 1,400 amount. This test initially stakes 7,000, enters a 80% cooldown and then stakes another 3,000 with the exit cooldown flag set to true.